pkg/render/manager_route_config.go

```diff
@@ -0,0 +1,416 @@
+// Copyright (c) 2024 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package render
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	voltronRoutesConfigMapName = "voltron-routes"
+	voltronRoutesFolderPath    = "/config_maps/voltron-routes"
+
+	configMapFolder = "/config_maps"
+	secretsFolder   = "/secrets"
+
+	uiTLSTerminatedRoutesKey              = "uiTLSTerminatedRoutes.json"
+	upstreamTunnelTLSTerminatedRoutesKey  = "upstreamTunnelTLSTerminatedRoutes.json"
+	upstreamTunnelTLSPassThroughRoutesKey = "upstreamTunnelTLSPassThroughRoutes.json"
+
+	routesAnnotationKey = "hash.operator.tigera.io/route-configuration"
+)
+
+type tlsTerminatedRoute struct {
```

Comment:
At first glance this struct seems to be very close if not identical to the `operatorv1.TLSTerminatedRoute` structure - I found myself wondering if there was really a need for two representations of this, until I realized that this structure is the Go binding for the JSON API that Voltron understands. 

I think it's worth calling that out explicitly at the top - i.e., "This structure is the Golang representation a single route within the JSON configuration file passed to Voltron."

Reply:
The fact that it has the same name is also really tripping me up when reading this code. We have:

- tlsTerminatedRoute
- v1.TLSTerminatedRoute
- voltronRouteConfigBuilder.tlsTerminatedRoutes (a []v1.TLSTerminatedRoute, not []tlsTerminatedRoute)

Reply:
> I think it's worth calling that out explicitly at the top - i.e., "This structure is the Golang representation a single route within the JSON configuration file passed to Voltron."

Ya this is exactly what it's for. I'll add some comments, but I'm not really sure about a better name for it.

Reply:
Added some comments.